src/config/index.ts

```diff
@@ -56,12 +63,31 @@ export const ENV_VARIABLES = {
 const environment: Environment =
   (ENV_VARIABLES.VITE_ENVIRONMENT as Environment | undefined) || 'mainnet';
 
+const isLocalServer =
+  (window.location.hostname === 'localhost' || window.location.hostname === '127.0.0.1') &&
+  environment !== 'ci' &&
+  environment !== 'storybook';
```

Comment:
I'd suggest using Vite's `DEV` environment variable for that, as it will cover all cases (localhost, but also if the dApp is running locally with the `--host` flag enable, in which case it would be hosted on an ip address):
```suggestion
const isLocalServer =
  import.meta.env.DEV &&
  environment !== 'ci' &&
  environment !== 'storybook';
```